docs/deployment/setup_gardener.md

```diff
@@ -1,44 +1,412 @@
+        - type: gvisor
+        name: containerd
+      version: 1592.3.0
+  machineTypes:
+  - architecture: amd64
+    cpu: "2"
+    gpu: "0"
+    memory: 4Gi
+    name: small_machine
+    storage:
+      class: standard
+      size: 64Gi
+      type: default
+    usable: true
+  regions:
+  - name: my-region-1
+    zones:
+    - name: my-region-1a
+    - name: my-region-1b
+  providerConfig:
+    apiVersion: openstack.provider.extensions.gardener.cloud/v1alpha1
+    kind: CloudProfileConfig
+    machineImages:
+    - name: gardenlinux
+      versions:
+      - version: 1592.3.0
+        regions:
+        - id: abcd-1234
+          name: my-region-1
+    # ...
+ ```
+
+Reference documentation:
+
+- [Cloud profiles](https://github.com/gardener/gardener/blob/master/docs/concepts/apiserver.md#cloudprofiles)
+- [Cloud profile example](https://github.com/gardener/gardener/blob/master/example/30-cloudprofile.yaml)
+- [Openstack cloud profile example](https://github.com/gardener/gardener-extension-provider-openstack/blob/master/docs/operations/operations.md)
+- [Garden Linux release notes example](https://github.com/gardenlinux/gardenlinux/releases/tag/1592.3)
+- [Image vector](https://github.com/gardener/gardener/blob/master/docs/deployment/image_vector.md)
+- [Image vector container defaults](https://github.com/gardener/gardener/blob/master/imagevector/containers.yaml)
+
+### DNS setup for internal & external domains
+
+Gardener maintains DNS records for a `Shoot` which requires a DNS zone and credentials to manage records in this zone. Provide the credentials alongside with information about the zone. The secrets are deployed to the virtual Garden cluster.
+
+Reference documentation:
+
+- [DNS records](https://github.com/gardener/gardener/blob/master/docs/extensions/resources/dnsrecord.md#what-does-gardener-create-dns-records-for)
+- [Default domain example](https://github.com/gardener/gardener/blob/master/example/10-secret-default-domain.yaml)
+- [Internal domain example](https://github.com/gardener/gardener/blob/master/example/10-secret-internal-domain.yaml)
+
+### Gardenlet
+
+Before the first `Shoot` can be created in a new Gardener landscape, at least one `Seed` is required. The very first `Seed` is typically created by deploying a properly configured `gardenlet` to an existing Kubernetes cluster. In this example, the `gardenlet` will be deployed to the runtime cluster - effectively turning it into a `Seed` as well.
+It is also possible (and strongly recommended for larger installations) to deploy a `gardenlet` to another cluster thus separating virtual Garden cluster and the first `Seed` with the control planes hosted there.
+
+For the `gardenlet` to function properly, it needs a `provider-extension` matching its configuration to create a `BackupBucket`, as well as valid infrastructure credentials. A `Secret` with the credentials needs to be created in the virtual Garden cluster and linked in the `gardenlet`'s configuration.
```

Comment:
Hm, again the backup should be optional. I am not sure, but I think this is not gardenlet but seed configuration (even though the 1:1 relation of seed and gardenlet, I think it is better to not describe it as gardenlet configuration).

Reply:
Addressed with a37e8e4b077f2b6ae111606124d82f0448377565